airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/format/format_command.py

```diff
@@ -75,16 +75,53 @@ def get_help_message(self) -> str:
             message = f"{message}."
         return message
 
-    def get_dir_to_format(self, dagger_client: dagger.Client) -> dagger.Directory:
-        """Get the directory to format according to the file_filter.
+    async def get_dir_to_format(self, dagger_client) -> Directory:
+        """Get a directory with all the source code to format according to the file_filter.
+        We mount the the files to format in a git container and remove all gitignored files.
+        It ensures we're not formatting files that are gitignored.
 
         Args:
-            dagger_client (dagger.Client): The dagger client to use to get the directory
+            dagger_client (dagger.Client): The dagger client to use to get the directory.
 
         Returns:
-            dagger.Directory: The directory to format
+            Directory: The directory with the files to format that are not gitignored.
         """
-        return dagger_client.host().directory(self.LOCAL_REPO_PATH, include=self.file_filter, exclude=DEFAULT_FORMAT_IGNORE_LIST)
+        # Load a directory from the host with all the files to format according to the file_filter and the .gitignore files
+        dir_to_format = dagger_client.host().directory(
+            self.LOCAL_REPO_PATH, include=self.file_filter + ["**/.gitignore", ".gitignore"], exclude=DEFAULT_FORMAT_IGNORE_LIST
+        )
+
+        return (
+            dagger_client.container()
+            .from_("alpine:latest")
+            .with_exec(
+                sh_dash_c(
+                    [
+                        "apk update",
+                        "apk add git",
+                    ]
+                )
+            )
+            .with_workdir(REPO_MOUNT_PATH)
+            .with_mounted_directory(REPO_MOUNT_PATH, dir_to_format)
+            # All with_exec commands below will re-run if the to_format directory changes
+            .with_exec(["git", "init"])
+            # This is needed to avoid git complaining about the user not being set
+            .with_exec(["git", "config", "user.email", "ci@airbyte.io"])
+            .with_exec(["git", "config", "user.name", "Airbyte CI"])
+            # Add all files to the git index
+            .with_exec(["git", "add", "."])
+            # Commit all files
+            .with_exec(["git", "commit", "--message", "Commit for formatting"])
+            # Remove all gitignored files
+            .with_exec(["git", "clean", "-dfX"])
+            # Delete all .gitignore files
+            .with_exec(sh_dash_c(['find . -type f -name ".gitignore" -exec rm {} \;']))
+            # Delete .git
+            .with_exec(["rm", "-rf", ".git"])
+            .directory(REPO_MOUNT_PATH)
```

Comment:
`.directory(REPO_MOUNT_PATH, include=self.file_filter)` and then you don't need to rm -rf .git and the .gitignore files?

Reply:
If you don't go down that route, note that the method docstring references `file_filter`.

Reply:
It's not possible to filter directories extracted from containers. Include and exclude options are only available when extracting a directory from the host.

Reply:
Oh, right, that's unfortunate.